universal_bridge.md

```diff
@@ -0,0 +1,456 @@
+    // TODO: should they be part of the message?
+    string calldata name,
+    string calldata symbol,
+    uint8 decimals
+) external returns (address token, uint256 amount) {
+    require(msg.sender == msgHeader.receiver, "Only receiver can claim");
+    require(msgHeader.chainDest == block.chainid, "Wrong destination chain");
+    require(msgHeader.label == "SEND", "Must read a SEND message")
+
+    // 1) Read and consume the SEND message
+    bytes memory m = mailbox.read(MessageHeader(sessionIDchainSrc, receiver, sessionId, "SEND");
+    if (m.length == 0) revert("No SEND message");
+
+    uint256 rmoteChainID;
+    address remoteAsset;
+
+    (remoteChainID, remoteAsset, amount) =
+        abi.decode(m, (uint256, address, uint256));
+
+
+    // 2) RELEASE if native token is hosted & escrowed here, else MINT BCT
+    if remoteChainID == block.chainid && IERC20(remoteAddress).balanceOf(address(this)) >= amount) {
+        // Release escrowed native tokens
+        require(IERC20(native).transfer(receiver, amount), "Native release failed");
+        token = native;
+    } else {
+        // Mint deterministic CET on this chain
+        token = ensureCETAndMint(remoteAddress, name, symbol, decimals, msgHeader.receiver, amount)
+    }
+
+    // 3) ACK back to source
+    mailbox.write(chainSrc, sender, sessionId, "ACK SEND", abi.encode("OK"));
+
+    emit TokensReceived(token, amount);
+    return (token, amount);
+}
+```
+
+------------------------------------------------------------------------
+
+### Safety & Replay
+
+-   **Mint authority:** CET enforces `onlyBridge` on `mint`/`burn`.
+-   **Replay protection:** `mailbox.read(...)` must be consume-once.
+-   **No registry needed:** CET address is computed deterministically
+    from L1 asset address.
+-   **ACK:** Ensures mailbox equivalency.
+
+------------------------------------------------------------------------
+
+###  End-to-End Lifecycle
+
+1.  Sender calls `bridgeERC20To` or `bridgeCETTo` with sessionId.
+2.  Source bridge locks/burns + writes `"SEND"` with L1 asset address.
+3.  Receiver calls `receiveTokens`.
+4.  Destination bridge reads `"SEND"`, computes CET address from
+    `remoteAsset`, mints, writes `"ACK SEND"`
+5.  ACK is visible for source monitoring.
+
+------------------------------------------------------------------------
+
+## L1 <-> L2 Bridge For native rollups.
+
+We need to utilize the current OP-contracts with minimal changes. Namely the `StandardBridge.sol`, `CrossDomainMessenger.sol`, and thei L1/L2 couterparts are neccessary. 
+
+Currently an OP rollup manage the L1<->L2 bridge via `OptimismPortal2` contract. This utilizes an `ETHLockbox` contract that locks all deposited ETH. Each native rollups using the universal bridge will deploy a `ComposePortal` (similar to `OptimismPortal)  that will use a shared `ETHLockBox` and an `ERC20LockBox`. The sharing of a single `LockBox` will ensure that funds deposited on any chain can be withdrawn via another chain.
```

Comment:
Question about the shared lockbox: do we need an accounting model per rollup? or just shared funds?
Maybe it's enough with a deposit proof?